core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -596,6 +596,92 @@ public CompletableFuture<Optional<Throwable>> acknowledge(String memberId, List<
         return CompletableFuture.completedFuture(Optional.ofNullable(throwable));
     }
 
+    /**
+     * Release the acquired records for the share partition. The next fetch offset is updated to the next offset
+     * that should be fetched from the leader.
+     *
+     * @param memberId The member id of the client that is fetching/acknowledging the record.
+     *
+     * @return A future which is completed when the records are released.
+     */
+    public CompletableFuture<Optional<Throwable>> releaseAcquiredRecords(String memberId) {
+        log.trace("Release acquired records request for share partition: {}-{}", groupId, topicIdPartition);
+
+        Throwable throwable = null;
+        lock.writeLock().lock();
+        List<InFlightState> updatedStates = new ArrayList<>();
+        try {
+            long localNextFetchOffset = nextFetchOffset;
+            // TODO: recordState can be ARCHIVED as well, but that will involve maxDeliveryCount which is not implemented so far
```

Comment:
I think there is a misunderstanding. `recordState` is the value that the acquired records should transition to. In case 
 `deliveryCount == group.share.delivery.attempt.limit`, then the `recordState` should be ARCHIVE not AVAILABLE. Since we do not have maxDeliveryCount implemented yet, that's why record can only transition from ACQUIRE to AVAILABLE and not ACQUIRE to ACKNOWLEDGE

Reply:
Sounds good.